infrastructure/core/network.tf

```diff
@@ -77,3 +77,46 @@ resource "azurerm_private_dns_zone_virtual_network_link" "all" {
     azurerm_private_dns_zone.all
   ]
 }
+
+resource "azurerm_network_security_group" "core" {
+  name                = "nsg-default-${var.naming_suffix}"
+  location            = azurerm_resource_group.core.location
+  resource_group_name = azurerm_resource_group.core.name
+
+  security_rule {
+    name                       = "deny-internet-outbound-override"
+    description                = "Blocks outbound internet traffic unless an explicit outbound-allow rule exists. Overrides the default rule 65001"
+    priority                   = 2000
+    access                     = "Deny"
+    protocol                   = "*"
+    direction                  = "Outbound"
+    destination_address_prefix = "Internet"
+    destination_port_range     = 443
+    source_address_prefix      = "*"
+    source_port_range          = "*"
+  }
+}
+
+# Network watcher must be added in real-data scenarios, thus will fail if network_watcher is undefined
+resource "azurerm_network_watcher_flow_log" "data_sources" {
+  count                     = (var.monitoring.network_watcher != null) || var.accesses_real_data ? 1 : 0
+  name                      = "nw-log-${var.naming_suffix}"
+  resource_group_name       = var.monitoring.network_watcher.resource_group_name
+  network_watcher_name      = var.monitoring.network_watcher.name
+  network_security_group_id = azurerm_network_security_group.core.id
+  storage_account_id        = azurerm_storage_account.core.id
+  enabled                   = true
```

Comment:
Is this not the default? I.e. does it need to be specified?

Reply:
it is, but explicit is good(?) and safeguards against the provider altering the default under us